pkg/resolver/topology.go

```diff
@@ -0,0 +1,122 @@
+package resolver
+
+import (
+	"fmt"
+
+	"github.com/redhat-appstudio/tssc-cli/pkg/config"
+)
+
+// Topology represents the dependency topology of charts. It is used to determine
+// the order in which charts should be installed.
+type Topology struct {
+	dependencies config.Dependencies // dependency topology
+}
+
+// GetDependencies returns the list of dependencies.
+func (t *Topology) GetDependencies() config.Dependencies {
+	return t.dependencies
+}
+
+// GetDependencyForChart returns the dependency for a given chart name.
+func (t *Topology) GetDependencyForChart(
+	name string,
+) (*config.Dependency, error) {
+	for i := range t.dependencies {
+		if t.dependencies[i].Chart.Name() == name {
+			return &t.dependencies[i], nil
+		}
+	}
+	return nil, fmt.Errorf("dependency not found for chart %s", name)
+}
+
+// exists checks if a dependency exists in the topology.
+func (t *Topology) exists(dependency config.Dependency) bool {
+	for _, d := range t.dependencies {
+		if d.Chart.Name() == dependency.Chart.Name() {
+			return true
+		}
+	}
+	return false
+}
+
+// PrependBefore prepends a list of dependencies before a specific chart.
+func (t *Topology) PrependBefore(name string, dependencies ...config.Dependency) {
+	prefix := config.Dependencies{}
+	for _, dependency := range dependencies {
+		if !t.exists(dependency) {
+			prefix = append(prefix, dependency)
+		}
+	}
+	if len(prefix) == 0 {
+		return
+	}
+
+	// Find the index where the chart name exists.
+	insertIndex := -1
+	for i, d := range t.dependencies {
+		if d.Chart.Name() == name {
+			insertIndex = i
+			break
+		}
+	}
+
+	// Insert the prefix slice before the found chart name index. If the chart is
+	// not found, prepend to the very beginning of the slice.
+	if insertIndex != -1 {
+		t.dependencies = append(
+			t.dependencies[:insertIndex],
+			append(prefix, t.dependencies[insertIndex:]...)...,
+		)
+	} else {
+		t.dependencies = append(prefix, t.dependencies...)
+	}
+}
+
+// AppendAfter inserts dependencies after a given chart name. If the chart does
+// not exist, it apends to the end the slice.
+func (t *Topology) AppendAfter(name string, dependencies ...config.Dependency) {
```

Comment:
Feels redundant as well. AddAfter or InsertAfter would be better.

Reply:
The same than [previous comment](https://github.com/redhat-appstudio/tssc-cli/pull/1120/files/953a39dccff1a060bb479e7313e05ee3f5bd4a12#r2267908644), to add on this, the "append" is a widely known function in Golang and there it's specialized to append after a given entry.